blog/topology-rules.md

```diff
@@ -0,0 +1,120 @@
+
+In this blog we will see two use cases, one is a network component  which is not part of skydive network. Secondly, a non network component.
+
+Before that, we will see some basic usage of Topology rules API.
+
+## Create Node
+
+To create node, You must provide a unique node name and a valid node type. You can provide some optional params as well.
+
+{% highlight shell %}
+skydive client node-rule create --action="create" --node-name="node1" --node-type="fabric" --name="node rule1"
+{
+  "UUID": "ea21c30f-cfaa-4f2d-693d-95159acb71ed",
+  "Name": "node rule1",
+  "Description": "",
+  "Metadata": {
+    "Name": "node1",
+    "Type": "fabric"
+  },
+  "Action": "create",
+  "Query": ""
+}
+{% end highlight %}
+
+## Update Nodes’ Metadata
+
+To update the metadata of an existing node, you must provide a ‘gremlin query’ to chose the nodes you want to update the metadata. According to your query you can update the metadata of one or more nodes with single node rule.
+
+{% highlight shell %}
+
+skydive client node-rule create --action="update" --name="update rule" --query="G.V().Has('Name', 'node1')" --metadata="key1=val1, key2=val2"
+{
+  "UUID": "3e6c0e15-a863-4583-6345-715053ac47ce",
+  "Name": "update rule",
+  "Description": "",
+  "Metadata": {
+    "key1": "val1",
+    "key2": "val2"
+  },
+  "Action": "update",
+  "Query": "G.V().Has('Name', 'node1')"
+}
+{% end highlight %}
+
+## Create Edge
+
+To create edge, you should provide the source and destination nodes AND type of edge, the value should be either `layer2` or `ownership`.
+
+{% highlight shell %}
+skydive client edge-rule create --name="edge" --src="G.v().has('TID', '2f6f9b99-82ef-5507-76b6-cbab28bda9cb')" --dst="G.V().Has('TID', 'd6ec6e2f-362e-51e5-4bb5-6ade37c2ca5c')" --relationtype="both"
+{
+  "UUID": "50fec124-c6d0-40c7-42a3-2ed8d5fbd410",
+  "Name": "edge",
+  "Description": "",
+  "Src": "G.v().has('TID', '2f6f9b99-82ef-5507-76b6-cbab28bda9cb')",
+  "Dst": "G.V().Has('TID', 'd6ec6e2f-362e-51e5-4bb5-6ade37c2ca5c')",
+  "Metadata": {
+    "RelationType": "both"
+  }
+}
+{% end highlight %}
+
+## USECASE 1
+
+In this use case, we will see how to add a network device which is not part of the skydive network. Let’s consider this example, we have two skydive agents running in two different hosts, to connect those two hosts we need a TOR switch. Even Though we can achieve this via defining a fabric nodes and links in config file, let’s see how we can do the same using Topology rules API.
+
+Without TOR switch, two agents will look like two different node without any links as shown in the below image.
+
+<p class="center">
+  <a href="/assets/images/blog/rules1.png" data-lightbox="Skydive Topology" data-title="Skydive topology view">
+    <img src="/assests/images/blog/rules1.png"/>
```

Comment:
you forgot to add the image or the link is broken

Reply:
done